index.js

```diff
@@ -67,11 +67,27 @@ WebRTCSwarm.prototype.close = function (cb) {
 }
 
 function setup (swarm, peer, id) {
+  var delayedStreams = [];
+
+  peer.on('stream', function onStreamWhileNotConnected (stream) {
+    debug('peer', id, 'is', (peer.connected ? '' : 'not ') + 'connected')
+    if (peer.connected) {
+      peer.removeListener('stream', onStreamWhileNotConnected)
+    } else {
+      debug('on stream while peer', id, 'not connected', stream)
+      delayedStreams.push(stream)
+    }
+  })
+
   peer.on('connect', function () {
     debug('connected to peer', id)
     swarm.peers.push(peer)
     swarm.emit('peer', peer, id)
     swarm.emit('connect', peer, id)
+    delayedStreams.forEach(function (stream) {
+      peer.emit('stream', stream)
+    })
+    delayedStreams = [];
```

Comment:
If I move this to line 75, the array will be only freed in case of receiving a stream after connecting. Actually, this cleaning is incomplete, I should remove also the event listener to avoid memory leaks and nullify the array. I'll do in the following commit.